docs: Proofreading + general improvements (#26442)
renovatebot/renovate: docs/usage/configuration-options.md

## branchNameStrict

> If `true`, Renovate removes special characters when slugify-ing the branch name:

Author: Maybe:
```suggestion
If `true`, Renovate removes special characters when it generates a slug for the branch name:
```
?

## branchTopic

This field is combined with `branchPrefix` and `additionalBranchPrefix` to form the full `branchName`. `branchName` uniqueness is important for dependency update grouping or non-grouping so be cautious about ever editing this field manually.
This is an advanced field, and it's recommended you seek a config review before applying it.

## bumpVersion

> Currently, this setting supports `helmv3`, `npm`, `nuget`, `maven`, `pep621`, `poetry` and `sbt` only, so raise a feature request if you have a use for it with other package managers.

Author: You're right. Feature requests are probably the old way?

}
```

> Another example would be if you want to configure a DCO sign-off to each commit.

Author: I think you're right here.

```dockerfile title="Dockerfile"
FROM amd64/ubuntu:18.04
ENV GRADLE_VERSION=6.2  # gradle-version/gradle&versioning=maven

> ENV NODE_VERSION=10.19.0  # github-tags/nodejs/node&versioning=node

Author: No, it's something I'm a fan of, having 2 spaces before starting a comment. It's something that stuck with me from https://peps.python.org/pep-0008/#inline-comments.

}
```

> To enable custom managers you'll need to add `custom.` prefix before their names

Author: ```suggestion
To enable custom managers you must add the `custom.` prefix before their names
```
Not sure if `the` fits better here?

- GitHub (.com and Enterprise Server)
- GitLab (.com and CE/EE)

> If you're running on any platform except `github.com`, you need to [configure a Personal Access Token](./getting-started/running.md#githubcom-token-for-changelogs) to allow Renovate to fetch changelogs notes from `github.com`.

Author: Works. I noticed it because since this was written the header changed to changelogs.

In this case you can probably keep using that default `fileMatch`.

At other times, the possible files is too vague for Renovate to have any default.

> For default, Kubernetes manifests can exist in any `*.yaml` file, and we don't want Renovate to parse every single YAML file in every repository just in case some of them have a Kubernetes manifest, so Renovate's default `fileMatch` for manager `kubernetes` is actually empty (`[]`) and needs the user to tell Renovate what directories/files to look in.

Author: Agree.

## ignorePrAuthor

This is usually needed if someone needs to migrate bot accounts, including from the Mend Renovate App to the self-hosted variant.
If `ignorePrAuthor` is configured to true, it means Renovate will fetch the entire list of repository PRs instead of optimizing to fetch only those PRs which it created itself.

> You should only want to enable this if you're changing the bot account (e.g. from `@old-bot` to `@new-bot`) and want `@new-bot` to find and update any existing PRs created by `@old-bot`.

Author: Maybe
```suggestion
You should only enable this if you're changing the bot account (e.g. from `@old-bot` to `@new-bot`) and want `@new-bot` to find and update any existing PRs created by `@old-bot`.
```

I think "want to" is what makes this weird.

## prPriority

Sometimes Renovate needs to rate limit its creation of PRs, e.g. hourly or concurrent PR limits.

> By default, Renovate sorts/prioritizes based on the update type, going from the smallest update to the biggest update.

Author: Not sure if I agree. Something for a native English speaker to decide?

}
```

> The field supports multiple URLs, but it is datasource-dependent on whether only the first is used or multiple.

Author: I agree, although I wouldn't know the answer to question 2 and 3. The answer to question 1 is the field "registryUrls", this text is part of that header.

## versioning

Usually, each language or package manager has a specific type of "versioning":

> JavaScript uses the npm SemVer implementation, Python uses pep440, etc.

Author: Maybe
```suggestion
JavaScript uses the npm version of SemVer, Python uses pep440, etc.
```
?
